### PR TITLE
Initial pass at using defaults for configuration

### PR DESCRIPTION
**Pull request by wscalf**
## Describe your changes

This greatly reduces the required fields and focuses on things that vary between environments.

## Ticket reference (if applicable)
Fixes #

## Checklist

* [x] Are the agreed upon acceptance criteria fulfilled?

* [x] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [x] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [x] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [x] Are the agreed upon coding/architectural practices applied?

* [x] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: bootstrap/app.go [hunk 1]
```diff
@@ -23,7 +23,38 @@ var waitForCompletion *sync.WaitGroup
 func getConfig(configPath string) (serviceconfig.ServiceConfig, error) {
 	cfg, err := NewConfigurationBuilder().
 		ConfigFilePath(configPath).
-		NoDefaults().
+		Defaults(serviceconfig.ServiceConfig{
```
Comment (merlante) [inline bootstrap/app.go:26]: LGTM

On a slight tangent, as a user, I still don't know what I *need* to specify in my config file as required config. I wonder if there is a way of specifying what the minimum config is? The defaults kind of say, "we have these defaults, so you don't have to worry about specifying these," but we must have some other config that neither has a default nor is it strictly required. (Or we will have soon.)

File: config.yaml [hunk 2]
```diff
@@ -1,40 +1,40 @@
-grpcPort: 50052 # GRPC port of the AuthZ service
-httpPort: 8081 # HTTP port of the Authz Service
-httpsPort: 8443 # HTTPS port of the AuthZ service
-logRequests: true # Request logging midddleware on/off - true/false
+#grpcPort: 50052 # GRPC port of the AuthZ service
```
Comment (DGuhr) [inline config.yaml:1]: I'd add a comment on top and in the readme (section: Configuration) that the commented out values are default values and don't need to be explicitly set until another value is needed.

**Review by DGuhr — APPROVED** [review]
LGTM, we can iterate on details in follow-up PRs i'd say. Not sure there's something to do downstream now before merging? cc @merlante @wscalf

**Comment by wscalf** [comment]
Not before merging, but we can remove part of the downstream config after.

File: bootstrap/app.go (hunk 1 above)
Comment (wscalf) [reply bootstrap/app.go:26]: Added a comment block that shows what the minimal values are.

**Pull request merged**